hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/rate/SerialConsumerRateLimiter.java

```diff
@@ -22,38 +24,45 @@
 
     private final OutputRateCalculator outputRateCalculator;
 
-    private final DeliveryCounters deliveryCounters = new DeliveryCounters();
+    private final SendCounters sendCounters;
 
     private OutputRateCalculator.Mode currentMode;
 
-    public SerialConsumerRateLimiter(Subscription subscription, OutputRateCalculator outputRateCalculator,
-                                     HermesMetrics hermesMetrics, ConsumerRateLimitSupervisor rateLimitSupervisor) {
-
+    public SerialConsumerRateLimiter(Subscription subscription,
+                                     OutputRateCalculatorFactory outputRateCalculatorFactory,
+                                     HermesMetrics hermesMetrics,
+                                     ConsumerRateLimitSupervisor rateLimitSupervisor,
+                                     Clock clock) {
         this.subscription = subscription;
         this.hermesMetrics = hermesMetrics;
         this.rateLimitSupervisor = rateLimitSupervisor;
-        this.outputRateCalculator = outputRateCalculator;
+        this.sendCounters = new SendCounters(clock);
+        this.outputRateCalculator = outputRateCalculatorFactory.createCalculator(subscription, sendCounters);
         this.currentMode = OutputRateCalculator.Mode.NORMAL;
         this.rateLimiter = RateLimiter.create(calculateInitialRate().rate());
         this.filterRateLimiter = RateLimiter.create(subscription.getSerialSubscriptionPolicy().getRate());
     }
 
     @Override
     public void initialize() {
+        outputRateCalculator.start();
         adjustConsumerRate();
-        hermesMetrics.registerOutputRateGauge(subscription.getTopicName(), subscription.getName(), rateLimiter::getRate);
+        hermesMetrics.registerOutputRateGauge(
```

Comment:
is there some unregister for this register? becasue if not, we might not be able restart subscription due to "metric already exists" error